src/move_ref.rs

```diff
@@ -321,13 +321,50 @@ unsafe impl<'a, T: ?Sized> DerefMove for MoveRef<'a, T> {
   }
 }
 
-unsafe impl<P> DerefMove for Pin<P>
+unsafe impl<'a, T> DerefMove for Pin<&'a T>
```

Comment:
No. It's because I had to change the signature of `mov` from:

```rust
pub fn mov<P>(ptr: impl Into<Pin<P>>) -> impl New<Output = P::Target>
```

to
```rust
pub fn mov<P>(ptr: P) -> impl New<Output = P::Target>
```

This in turn is because `cxx::UniquePtr` is _intrinsically_ pinned and does not need to be wrapped in `Pin`, which I know you don't really approve of, but there we go :)

For all previously supported types of pointers, that means that `Pin<P>` needs to be `DerefMove` rather than `P`. This is, of course, a significant change to the meaning of `DerefMove`.

I tried to add a blanket implementation here for all `Into<Pin<P>>`, but (as you might remember) I couldn't make it work. Hence, instead, implementing `DerefMove` for every type of pointer.